adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapter.java

```diff
@@ -451,6 +461,27 @@ private void onAuthenticationSuccess(final MqttEndpoint endpoint, final Device a
         return result;
     }
 
+    private void handlePublishedMessage(final MqttContext context) {
+        // there is is no way to extract a SpanContext from an MQTT 3.1 message
+        // so we start a new one for every message
+        final Span span = tracer.buildSpan("upload message")
+            .ignoreActiveSpan()
+            .withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_SERVER)
+            .withTag(Tags.MESSAGE_BUS_DESTINATION.getKey(), context.message().topicName())
+            .withTag(TracingHelper.TAG_QOS.getKey(), context.message().qosLevel().toString())
+            .withTag(Tags.COMPONENT.getKey(), getTypeName())
+            .start();
+        context.put(KEY_CURRENT_SPAN, span);
+        onPublishedMessage(context).setHandler(processing -> {
+            if (processing.succeeded()) {
+                span.setTag(Tags.HTTP_STATUS.getKey(), HttpURLConnection.HTTP_ACCEPTED);
```

Comment:
Just a minor thing, isn't it weird to read "http.status_code" (the `HTTP_STATUS` value) in a trace related to MQTT request? Can we add a custom MQTT tag for that?

Reply:
In general, I agree. However, the OpenTracing documentation advises to use the standard tags wherever possible. Since we are indeed using HTTP status code (semantics) all over Hono, I thought it would make sense to use this standard tag as well so it is clear that the code indeed has HTTP status code semantics. WDYT?

Reply:
I'm fine with that, just hoping that the tracing reader won't be get confused about it. Let's stick with it.